### ISSUE TITLE
False parser_error for adjacent literal in CONCATENATE (SAP accepts with warning)

### ISSUE THREAD

**Issue by melinerunen** [abaplint/abaplint#4322]
### Summary

The literal-separator lexer change from #4289 appears too strict for a `CONCATENATE` operand. On an on-premise ABAP 7.50 system, the statement below passes syntax check with one warning and executes correctly. In abaplint 2.120.58 it becomes a `parser_error`. This differs from the named-parameter case in #4289, which SAP rejects.

### Minimal reproduction

`src/zliteral_adjacent.prog.abap`:

```abap
REPORT zliteral_adjacent.
DATA lv_suffix TYPE string VALUE 'bar'.
DATA lv_result TYPE string.
CONCATENATE 'foo'lv_suffix INTO lv_result.
WRITE lv_result.
```

With `src/zliteral_adjacent.prog.xml`:

```xml
<?xml version="1.0" encoding="utf-8"?>
<abapGit><asx:abap xmlns:asx="http://www.sap.com/abapxml"><asx:values><PROGDIR><NAME>ZLITERAL_ADJACENT</NAME></PROGDIR></asx:values></asx:abap></abapGit>
```

And `abaplint.json`:

```json
{
  "global": { "files": "/src/**/*.*" },
  "syntax": { "version": "v750", "errorNamespace": "^(Y|Z)" },
  "rules": { "parser_error": true }
}
```

### Observed

- SAP syntax check: one **warning** at the `CONCATENATE` line saying a space or equivalent character is needed after `'foo'`; no syntax error. The report runs and `lv_result` is `foobar`.
- abaplint 2.120.50: 0 issues, 2 files analyzed.
- abaplint 2.120.58: `zliteral_adjacent.prog.abap[4, 1] - Statement does not exist in the configured ABAP version(or a parser error), "CONCATENATE" (parser_error) [E]`.

### Expected

The parser accepts this executable statement. A separate warning is reasonable, but `parser_error` treats it as unparsable and can suppress AST-derived facts from the statement.

Reproduced with the published CLI versions and independently checked in SAP. This report was prepared with AI assistance; the SAP syntax-check and runtime observations were supplied by an ABAP developer.

**Comment by larshp** [comment]
thanks, workaround is adding a space
